src/Sqids.php

```diff
@@ -655,11 +656,12 @@ public function encode(array $numbers): string
             return '';
         }
 
-        $inRangeNumbers = array_filter($numbers, fn($n) => $n >= 0 && $n <= self::maxValue());
-        if (count($inRangeNumbers) != count($numbers)) {
-            throw new InvalidArgumentException(
-                'Encoding supports numbers between 0 and ' . self::maxValue(),
-            );
```

Comment:
Instead of creating a new array, the exception is thrown directly when there is an invalid number.